### DISCUSSION TITLE
Basic File encryption/decrypt

### DISCUSSION THREAD

**Discussion by vHanda** [GitJournal/GitJournal#806] [Feature Requests]
Allow files to be encrypted and stored in the Git Repo as binary blobs.

The idea is to use standard AES encryption which can also be done via the [openssl command line](https://medium.com/@kekayan/encrypt-files-using-aes-with-openssl-dabb86d5b748).

Edit: Reasons why this might not be a good idea - https://stackoverflow.com/posts/16056298/revisions

---
> Please vote on this issue with a 👍reaction. It's one of the primary mechanisms used to prioritize issues. Please also subscribe to the issue if you're interested, as that way I can discuss possible implementations and share mockups.

**Comment by mipro98** [comment]
I _personally_ would prefer not to store binary blobs in a git repository. I'm currently not that much into encryption but wouldn't it be possible to preserve a file to stay plain-text but encrypted (symmetrically)? At least that's possible with GPG, afaik.

**Comment by vHanda** [comment]
> I _personally_ would prefer not to store binary blobs in a git repository. I'm currently not that much into encryption but wouldn't it be possible to preserve a file to stay plain-text but encrypted (symmetrically)? At least that's possible with GPG, afaik.

Hi. Could you please give me an example? I don't understand how the note could be encrypted and be in plain-text.

**Comment by mipro98** [comment]
GPG has a `--armor` flag which lets you encrypt files / text in ASCII format. You can easily try it out yourself:

```bash
# encrypt
gpg --symmetric --armor your-textfile.md
# decrypt
gpg --decrypt your-textfile.md.asc
```

Of course this also works using asymmetrical encryption.

For example the following text is encrypted with the key `GitJournal`:

```
-----BEGIN PGP MESSAGE-----

jA0ECQMCQNkPoGNZBe//0o4BTWDbL2KKthSljs9Xhc+HyqBevMMjYN0NQFjOuZvO
Qjo5iYfMyCorhuV72gn9VoIhuAjoXR3DZDQL5a7Fim+IGkwYdOX6/dtXc8bPhhyv
J/fX5XJON9TDNrNiZ9koUxK0y0zJovMShVueu+BcBSMlIyNDQ+BJIuxHxsUBGd/E
pynF556dAAdda8+6Fwld
=eBpp
-----END PGP MESSAGE-----
```

However, although it is not a binary file, its content would change completely every time the file gets changed which would also result in larger space occupation in the repository just as with binary blobs. Maybe it is still better, i don't know. At least it would be readable by any text editor outside of GitJournal (e.g. for copy paste where you cannot download the entire file in some edge-cases, e.g. when viewing the file on github).

**Comment by vHanda** [comment]
I've started using [git-crypt](https://github.com/AGWA/git-crypt) in some other projects, and I quite like how it works. For now I would prefer to implement support for - https://github.com/GitJournal/GitJournal/issues/291 - over a custom encryption mechanism. Even though git-crypt results in blobs.
